### PR TITLE
Documented improvements to RKE2 Etcd Backup and Restore Docs

### PR DESCRIPTION
**Pull request by jtravee**
Closes https://github.com/rancher/docs/issues/3613 in rancher/docs.

1. Documented new step-by-step guides for restore to new & existing nodes.
2. Added notes on version-specific caveats into backup & restore.
3. Added new section for additional notes for restore.

Reference: https://docs.rke2.io/backup_restore/

### CODE REVIEW

File: docs/backup_restore.md [hunk 1]
```diff
@@ -27,23 +27,93 @@ rke2 server --cluster-reset
...
+> * By default, snapshots are enabled and are scheduled to be taken every 12 hours. The snapshots are written to `${data-dir}/server/db/snapshots` with the default `${data-dir}` being `/var/lib/rancher/rke2`.
+
+> **Version-specific requirement for rke2 v1.20.11+rke2r1**
+
+> * When restoring RKE2 from backup to a new node in rke2 v1.20.11+rke2r1, you should ensure that all pods are stopped following the initial restore by running `rke2-killall.sh` as follows:
+```
+curl -sfL https://get.rke2.io | sudo INSTALL_RKE2_VERSION=v1.20.11+rke2r1
+rke2 server \
+ --cluster-reset \
+ --cluster-reset-restore-path=<PATH-TO-SNAPSHOT> --token <token used in the original cluster>
```
Comment (brandond) [inline docs/backup_restore.md:108]:
```suggestion
 --cluster-reset-restore-path=<PATH-TO-SNAPSHOT> \
 --token=<token used in the original cluster>
```

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM; one nit on flag consistency

File: docs/backup_restore.md (hunk 1 above)
Comment (jtravee) [reply docs/backup_restore.md]: Thanks!

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]
LGTM

**Pull request merged**